cli/sign.go

```diff
@@ -0,0 +1,44 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/sigstore/cosign/cmd/cosign/cli/options"
+	"github.com/sigstore/cosign/cmd/cosign/cli/sign"
+
+	// Loads OIDC providers
+	_ "github.com/sigstore/cosign/pkg/providers/all"
+)
+
+// Sign applies keyless OIDC signatures to sign UOR Collections
+func Sign(ctx context.Context, dest string) error {
+
+	ko := options.KeyOpts{
+		RekorURL:        "https://rekor.sigstore.dev",
+		OIDCClientID:    "sigstore",
+		OIDCRedirectURL: "",
+		OIDCIssuer:      "https://oauth2.sigstore.dev/auth",
+		FulcioURL:       "https://fulcio.sigstore.dev",
+	}
+	os.Setenv("COSIGN_EXPERIMENTAL", "1")
+
+	regopts := options.RegistryOptions{
```

Comment:
The `RegistryOptions`  are using `go-containerregistry` under the hood and it has the `authn.Keychain` type exposed, so we will have to configure `authn.Keychains`. The default keychain correlates to the default configs we use, but we allow for custom configs to be used through the `--configs` flags. https://github.com/uor-framework/uor-client-go/blob/006ab2fa0eb7fc526d6058b821655e2ece58ac65/registryclient/orasclient/credentials.go#L105